### DISCUSSION TITLE
How to use absurd with E.fold

### DISCUSSION THREAD

**Discussion by adinvadim** [gcanti/fp-ts#1622] [Q&A] [answered]
I want to use construction like this.
```typescript
      pipe(
        data,
        E.fold(absurd, response => {
          expect(response.uid).toBe('uid');
          expect(response.email).toBe('mail@mail.mail');
          expect(response.role).toBe('user');
          verify(mockedFirebaseAuth.getUser('uid')).called();
          verify(
            mockedFirebaseAuth.setCustomUserClaims('uid', anything()),
          ).called();
        }),
      );
```

But TS error on typing 
```Argument of type '<A>(_: never) => A' is not assignable to parameter of type '(e: PersistenceError) => unknown'.```

My answer is: Is there possibility to use `absurd` in `fold`?

**Comment by adinvadim** [comment]
I used often this construction in test, but after more strict changes in tsconfig has this errors.
Now my alternative is
```typescript
      expect(E.isRight(data)).toBe(true);
      if (E.isRight(data)) {
        expect(data.right.uid).toBe('uid');
        expect(data.right.email).toBe('mail@mail.mail');
        expect(data.right.role).toBe('user');
        verify(mockedFirebaseAuth.getUser('uid')).called();
        verify(
          mockedFirebaseAuth.setCustomUserClaims('uid', anything()),
        ).called();
      }
```

But sometimes I like more to use `absurd` in `fold`

**Comment by samhh** [comment, marked as answer]
You can test the shape of the whole thing inside the constructor:

```ts
expect(data).toEqual(E.right({
  uid: 'uid',
  email: 'mail@mail.mail',
  role: 'user',
}))
```

Or use something like [unsafeUnwrap](https://samhh.github.io/fp-ts-std/modules/Either.ts.html#unsafeunwrap), which is mildly more ergonomic than `isRight`; throwing an error is fine in the context of testing.
